From 06c772d4b01994aae68cf694aa3b3d1cffaf06fa Mon Sep 17 00:00:00 2001
From: Timple Raj M <quic_timple@quicinc.com>
Date: Tue, 30 Jan 2024 20:58:01 +0530
Subject: [PATCH] drivers: clk: disable unused clk functions for 16M profile

disable clk-gate, clk-multiplier, clk-mux, clk-composite,
clk-fractional-divider and clk-gpio drivers.

This patch is snapshot of the commit
5e7b9a074f0616771b02d09ef2c1b33a75abfcb0
from win.linuxopenwrt.2.1 branch and
oss/kernel/linux-ipq-5.4 project

Change-Id: I47f12c033246b51f270504de8dac0fcbff72cca7
Signed-off-by: Timple Raj M <quic_timple@quicinc.com>
---
 drivers/clk/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/Makefile b/drivers/clk/Makefile
index 18969cbd4bb1..6bdce5620938 100644
--- a/drivers/clk/Makefile
+++ b/drivers/clk/Makefile
@@ -6,13 +6,15 @@ obj-$(CONFIG_CLK_KUNIT_TEST)	+= clk_test.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-divider.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-fixed-factor.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-fixed-rate.o
-obj-$(CONFIG_COMMON_CLK)	+= clk-gate.o
 obj-$(CONFIG_CLK_GATE_KUNIT_TEST) += clk-gate_test.o
+ifneq ($(CONFIG_IPQ_FLASH_16M_PROFILE), y)
+obj-$(CONFIG_COMMON_CLK)	+= clk-gate.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-multiplier.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-mux.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-composite.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-fractional-divider.o
 obj-$(CONFIG_COMMON_CLK)	+= clk-gpio.o
+endif
 ifeq ($(CONFIG_OF), y)
 obj-$(CONFIG_COMMON_CLK)	+= clk-conf.o
 endif
-- 
2.34.1

